api-admin/src/main/java/bio/terra/pearl/api/admin/service/AuthUtilService.java

```diff
@@ -76,6 +74,28 @@ public Portal authUserToPortal(AdminUser user, String portalShortcode) {
     throw new NotFoundException("Portal %s not found".formatted(portalShortcode));
   }
 
+  public Portal authUserToPortalWithPermission(
+      AdminUser user, String portalShortcode, String permission) {
+    Portal portal = authUserToPortal(user, portalShortcode);
+    if (user.isSuperuser()) {
+      return portal;
+    }
+    adminUserService
+        .findByUsernameWithPermissions(user.getUsername())
+        .ifPresent(
```

Comment:
if no user is found, this auth check will succeed. Granted, it's unlikely the user will have been deleted in between the time the AdminUser was fetched and this call, but this is auth code so it's better to be safe.  let's use ifPresentOrElse to throw an exception if there's no user found